pkg/driver/audiotest/dummy.go

```diff
@@ -81,7 +89,7 @@ func (d *dummy) AudioRecord(p prop.Media) (audio.Reader, error) {
 		}
 		return a, func() {}, nil
 	})
-	return reader, nil
+	return reader, d.state.Update(driver.StateRunning, func() error { return nil })
```

Comment:
Is it safe to allow changing state between `CheckUpdate` and `Update`, then return `invalid state` error?
Maybe it doesn't happen in typical usages?